redfish-core/lib/processor.hpp

```diff
@@ -1309,6 +1554,326 @@ inline void requestRoutesProcessor(App& app)
+                                 "Decorator.OperationalStatus")
+                {
+                    for (const auto& [proName, proValue] : properties)
+                    {
+                        if (proName == "Functional")
+                        {
+                            const bool* value = std::get_if<bool>(&proValue);
+                            if (value == nullptr)
+                            {
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+                            functional = *value;
+                        }
+                    }
+                }
+                else if (interface == "xyz.openbmc_project.Inventory.Item")
+                {
+                    for (const auto& [proName, proValue] : properties)
+                    {
+                        if (proName == "Present")
+                        {
+                            const bool* value = std::get_if<bool>(&proValue);
+                            if (value == nullptr)
+                            {
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+                            present = *value;
+                        }
+                        else if (proName == "PrettyName")
+                        {
+                            const std::string* prettyName =
+                                std::get_if<std::string>(&proValue);
+                            if (prettyName == nullptr)
+                            {
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+                            aResp->res.jsonValue["Name"] = *prettyName;
+                        }
+                    }
+                }
+                else if (interface ==
+                         "xyz.openbmc_project.State.Decorator.Availability")
+                {
+                    for (const auto& [proName, proValue] : properties)
+                    {
+                        if (proName == "Available")
+                        {
+                            const bool* value = std::get_if<bool>(&proValue);
+                            if (value == nullptr)
+                            {
+                                messages::internalError(aResp->res);
+                                return;
+                            }
+                            available = *value;
+                        }
+                    }
+                }
+            }
+
+            if (!available)
+            {
+                aResp->res.jsonValue["Status"]["State"] = "UnavailableOffline";
+            }
+            else if (!present)
+            {
+                aResp->res.jsonValue["Status"]["State"] = "Absent";
+            }
+            else if (!functional)
```

Comment:
This should be an if like have today 
https://github.com/ibm-openbmc/bmcweb/blob/1030/redfish-core/lib/processor.hpp#L1084